extra/modules/greenbids-real-time-data/src/test/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/FilterServiceTest.java

```diff
@@ -0,0 +1,137 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.v1;
+
+import ai.onnxruntime.OnnxTensor;
+import ai.onnxruntime.OnnxValue;
+import ai.onnxruntime.OrtException;
+import ai.onnxruntime.OrtSession;
+import ai.onnxruntime.TensorInfo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.data.ThrottlingMessage;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.FilterService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.OnnxModelRunner;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class FilterServiceTest {
+
+    @Mock
+    private OnnxModelRunner onnxModelRunnerMock;
+
+    @Mock
+    private OrtSession.Result results;
+
+    @Mock
+    private OnnxTensor onnxTensor;
+
+    @Mock
+    private TensorInfo tensorInfo;
+
+    private FilterService target;
+
+    @BeforeEach
+    public void setUp() {
+        target = new FilterService();
+    }
+
+    @Test
+    public void shouldReturnFilteredBiddersWhenValidThrottlingMessagesProvided() throws OrtException, IOException {
+        // given
+        final List<ThrottlingMessage> throttlingMessages = createThrottlingMessages();
+        final Double threshold = 0.5;
+        final OnnxModelRunner onnxModelRunner = givenOnnxModelRunner();
+
+        // when
+        final Map<String, Map<String, Boolean>> impsBiddersFilterMap = target.filterBidders(
+                onnxModelRunner, throttlingMessages, threshold);
+
+        // then
+        assertThat(impsBiddersFilterMap).isNotNull();
+        assertThat(impsBiddersFilterMap.get("adUnit1").get("bidder1")).isTrue();
```

Comment:
please use a few messages case, and cover the false case as well

Reply:
added more cases to throttlingMessages for filtered and kept bidders + asserts